src/main/scala/org/constellation/concurrency/cuckoo/CuckooFilter.scala

```diff
@@ -0,0 +1,25 @@
+package org.constellation.concurrency.cuckoo
+
+import java.io.ByteArrayInputStream
+
+import net.cinnom.nanocuckoo.NanoCuckooFilter
+import org.constellation.schema.snapshot.FilterData
+
+case class CuckooFilter(filterData: FilterData) {
+
+  private lazy val filter: NanoCuckooFilter = {
+    val stream = new ByteArrayInputStream(filterData.contents)
+    try {
+      val resultingFilter = new NanoCuckooFilter.Builder(filterData.capacity).build()
+      resultingFilter.readMemory(stream)
+      resultingFilter
+    } finally stream.close()
+  }
+
+  def contains[T](item: T)(implicit convert: T => String): Boolean = convert.andThen(filter.contains)(item)
+
+}
+
+object CuckooFilter {
+  def apply(filterData: FilterData): CuckooFilter = CuckooFilter(filterData)
```

Comment:
If `CuckooFilter` is a case class then it does nothing then it seems to be redundant to create explicit `apply`. Or maybe I am missing something 🤔

Reply:
:+1: looks like the apply method looks exactly the same as the built-in apply

Reply:
You're right guys. That class used not to be a case class and that's why the companion object was created.